File: codes/ExpressionTree/Python/ExpressionTreeMain.py
import BinaryTree2Module as b2m
import ExpressionTreeModule as etm

exp = "12+7*"
eTree = etm.ExpressionTree.MakeExpTree(exp)

print("전위 표기법 수식: ", end='')
etm.ExpressionTree.ShowPrefixTypeExp(eTree)
print()

print("중위 표기법 수식: ", end='')
etm.ExpressionTree.ShowInfixTypeExp(eTree)
print()

print("후위 표기법 수식: ", end='')
etm.ExpressionTree.ShowPostfixTypeExp(eTree)
print()

print("연산의 결과: {}".format(etm.ExpressionTree.EvaluateTree(eTree)))

File: codes/ExpressionTree/Java/ExpressionTree/ExpressionTreeMain.java
package ExpressionTree;

public class ExpressionTreeMain {
    public static void main(String[] args) {
        char exp[] = "12+7*".toCharArray();
        BinaryTree2<Character> eTree = ExpressionTree.MakeExpTree(exp);

        System.out.print("전위 표기법 수식: ");
        ExpressionTree.ShowPrefixTypeExp(eTree);
        System.out.println();

        System.out.print("중위 표기법 수식: ");
        ExpressionTree.ShowInfixTypeExp(eTree);
        System.out.println();

        System.out.print("후위 표기법 수식: ");
        ExpressionTree.ShowPostfixTypeExp(eTree);
        System.out.println();

        System.out.println("연산의 결과: "+ExpressionTree.EvaluateTree(eTree));
    }
}
